### PR TITLE
feat: allow uploading & downloading progress-ink data

### LINKED ISSUE #18: Let artists download and upload in-progress ink data
"In-progress ink data" is the json of lines that react-canvas-draw can render.

As an artist I want to be able to download this so I don't lose in-progress work
As an artist I want to be able to upload this to the canvas

_This is also aiming to better support the output of PythonAutoNifty_ cc @niftymaestro

### PR DESCRIPTION
**Pull request by sadda11asm**
closes #18

### CODE REVIEW

File: packages/react-app/src/CreateInk.js [hunk 1]
```diff
@@ -320,370 +322,417 @@ useEffect(() => {
...
+                  //setLoadedLines()
+                  props.setDrawing()
+                }}
+                okText="Yes"
+                cancelText="No"
+            >
+              <Button
+              ><ClearOutlined /> CLEAR</Button>
+            </Popconfirm>
+            <Button
```
Comment (azf20) [inline packages/react-app/src/CreateInk.js:536]: We may want to move these to be next to the "Drafts" buttons (see https://github.com/BuidlGuidl/nifty-ink/pull/35)
I think probably doesn't make sense mixed in with all the drawing tools?

File: packages/react-app/src/CreateInk.js [hunk 2]
```diff
@@ -320,370 +322,417 @@ useEffect(() => {
...
+    }
+  };
 
-  if (drawingCanvas.current.lines[drawingCanvas.current.lines.length - 1].ref) {
-    drawingCanvas.current.lines[0].brushColor = drawingCanvas.current.lines[drawingCanvas.current.lines.length - 1].brushColor;
-    let lines = drawingCanvas.current.lines.slice(0, -1);
-    triggerOnChange(lines);
-  } else {
-    let lines = drawingCanvas.current.lines.slice(0, -1);
+  const downloadCanvas = async () => {
```
Comment (azf20) [inline packages/react-app/src/CreateInk.js:359]: Nice!

File: packages/react-app/src/CreateInk.js [hunk 3]
```diff
@@ -320,370 +322,417 @@ useEffect(() => {
...
+    const myData = drawingCanvas.current.lines; // I am assuming that "this.state.myData"
+    // is an object and I wrote it to file as
+    // json
+    const fileName = "canvas";
```
Comment (azf20) [inline packages/react-app/src/CreateInk.js]: could we maybe include the timestamp here so users can download different files?

**Review by azf20 — CHANGES_REQUESTED** [review]
Looks good! I think makes sense to create a section with this and the "Drafts" buttons, as they kind of go together. Small comment on timestamp in the file. Nice work!

**Review by davidryan59 — APPROVED** [review]
I've tested it, works nicely!

Screenshot 2021-12-10 at 09 14 18

Approved, but a couple of changes would enhance further:
- Put a tooltip on DOWNLOAD button, like the existing one on UPLOAD
- Make the file name timestamped, and referring to Nifty Ink, such as `nifty_ink_canvas_YYYY_MM_DD_HH_mm_SS.json`

**Comment by sadda11asm** [comment]
fixed all the issues :)

**Review by azf20 — APPROVED** [review]

**Pull request merged**